packages/client/hooks/useResultsHeight.ts

```diff
@@ -1,21 +1,23 @@
 import {RefObject, useLayoutEffect, useState} from 'react'
 import useResizeObserver from './useResizeObserver'
 
-// if results are remotely ungrouped, SpotlightGroups increases in height.
+// if results are remotely ungrouped, SpotlightResults increases in height.
 // to prevent the modal height from changing, use initial groups height
-const useResultsHeight = (resultsRef: RefObject<HTMLDivElement>, spotlightSearchQuery: string) => {
+const useResultsHeight = (resultsRef: RefObject<HTMLDivElement>) => {
   const [height, setHeight] = useState<number | string>('100%')
 
   useLayoutEffect(() => {
     const newHeight = resultsRef.current?.clientHeight
-    if (newHeight && height !== newHeight && !spotlightSearchQuery) {
+    if (newHeight && height !== newHeight) {
       setHeight(newHeight)
     }
   }, [height])
 
   useResizeObserver(() => {
-    // when resized, set height to 100% so that useLayoutEffect can calc the new height
-    setHeight('100%')
+    // when resized & has ref, set height to 100% so that useLayoutEffect can calc the new height
+    if (resultsRef.current) {
```

Comment:
If there's no `resultsRef`, e.g. when we're showing the empty state, the height was being set to `100%` but not being recalculated in `useLayoutEffect` as `newHeight` was `undefined`